python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
It looks up the _DIR env vars and adds them to cfgPath as second time when recursing through dependencies.  https://github.com/lsst/sconsUtils/blob/master/python/lsst/sconsUtils/dependencies.py#L488

If you step through it with a debugger you'll see that every entry in cfgPath is duplicated.